src/main/capi/prepared-c.cpp

```diff
@@ -159,6 +159,49 @@ duckdb_logical_type duckdb_param_logical_type(duckdb_prepared_statement prepared
 	return nullptr;
 }
 
+idx_t duckdb_nfields(duckdb_prepared_statement statement) {
```

Comment:
Hi, @fanyang01! Could you update me on the difference/reasoning between these new functions? Are they (in a way) the equivalent of the other (just above in this file) functions post-execution? If so, should we make this more explicit in the naming? E.g., `duckdb_params -> duckdb_result_params` or so.

Reply:
They provide metadata about the return types of a statement. For example:  

```sql
-- CREATE TABLE t (id int, v double, s varchar);
SELECT id, v, s FROM t WHERE id = ?;
```  

- `duckdb_nfields` returns `3`, indicating the query returns three fields, while the existing `duckdb_nparams` returns `1`, indicating there is one parameter in the query. 
- `duckdb_field_name(0)` returns `"id"`, `duckdb_field_name(1)` returns `"v"`, and `duckdb_field_name(2)` returns `"s"`, corresponding to the names of the selected columns.  
- ...

Reply:
I'm unsure whether to refer to the returned columns as `field` or `column`. What's your opinion?

Reply:
Done! Thanks for suggesting the function names :)